web3/middleware/validation.py

```diff
@@ -15,14 +15,14 @@
 
 @curry
 def validate_chain_id(web3, chain_id):
-    if chain_id == web3.version.network:
+    if chain_id is None:
```

Comment:
what?? This is going away?? Just because this middleware I spent the last two hours trying to figure out https://github.com/voith/web3.py/blob/b3a30a7374cb5f205d7e96a326533dc8394c7a73/web3/utils/formatters.py#L168-L180.
I would have got #756 right the first time I did it if I knew this change was coming.

Also you might want to remove `web3` from `validate_chain_id(web3, chain_id)` as its no longer needed

Reply:
I guess the clearer thing to do is his to set it to:

```py
if chain_id == web3.net.chainId:
```

Right now, that's the same thing. But once we get nodes to make `chainId` available in a standard way, it should start working again.

I'll review #756 today.